File: LC/Medium/539.py
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        minutes = [int(time[:2]) * 60 + int(time[3:]) for time in timePoints]
        minutes.sort()
        ans = inf
        for i in range(len(minutes)-1):
            ans = min(ans, minutes[i+1] - minutes[i])
        
        return min(ans, 24 * 60 - minutes[-1] + minutes[0])

File: LC/C++/Medium/539.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        vector<bool> seen(1440, false);
        for (string& time : timePoints) {
            int minutes = stoi(time.substr(0, 2)) * 60 + stoi(time.substr(3, 5));
            if (seen[minutes]) return 0;
            seen[minutes] = true;
        }

        int first = -1, prev = -1, ans = INT_MAX;
        for (int i = 0; i < 24*60; ++i) {
            if (seen[i]) {
                if (first == -1) first = i;
                
                if (prev != -1) ans = min(ans, i - prev);
                prev = i;
            }
        }

        return min(ans, 24*60 - prev + first);
    }
};